tools/generate-hash.mjs

```diff
@@ -0,0 +1,99 @@
+import fs from 'fs/promises';
+import hasha from 'hasha';
+import minimatch from 'minimatch';
+import shell from 'shelljs';
+
+const options = {
+  algorithm: 'sha256',
+};
+
+/**
+ * @param {string} managerName
+ * @param {string} fileAddr
+ * @returns {Promise<string|undefined>}
+ */
+async function getFileHash(managerName, fileAddr) {
+  try {
+    const hash = await hasha.fromFile(
+      `./lib/modules/manager/${managerName}/${fileAddr}`,
+      options
+    );
+    return hash;
+  } catch (err) {
+    shell.echo(`ERROR: Unable to generate hash for ${managerName}/${fileAddr}`);
+    return undefined;
```

Comment:
Should we fail? Otherwise we can get silent missing hashes?

Reply:
If we get missing hashes, the Map will remain empty. And I'm planning that an empty Map or a Map miss would mean "cache is invalid" every time, i.e. it would gracefully fail and retry everything.

I think we want to fail the build process if this happens though, so that we hopefully never publish such a thing to npm.

Reply:
> If we get missing hashes, the Map will remain empty. And I'm planning that an empty Map or a Map miss would mean "cache is invalid" every time, i.e. it would gracefully fail and retry everything.
> 
> I think we want to fail the build process if this happens though, so that we hopefully never publish such a thing to npm.

Yes, that's what I think. Do fail publish in missing hashes

Reply:
@RahulGautamSingh can you make sure it fails with a non zero code of no hashes?